op-chain-ops/deployer/deployer.go

```diff
@@ -47,60 +49,23 @@ type Deployment struct {
-// the latest L1 hardforks enabled.
-// The returned backend should be closed after use.
-func NewL1Backend() (*backends.SimulatedBackend, error) {
-	backend, err := NewBackendWithGenesisTimestamp(ChainID, 0, true, nil)
-	return backend, err
-}
-
-// NewL2Backend returns a SimulatedBackend suitable for L2.
+// NewBackend returns a SimulatedBackend suitable for L2.
 // It has the latest L2 hardforks enabled.
 // The returned backend should be closed after use.
-func NewL2Backend() (*backends.SimulatedBackend, error) {
-	backend, err := NewBackendWithGenesisTimestamp(ChainID, 0, false, nil)
+func NewBackend() (*backends.SimulatedBackend, error) {
+	backend, err := NewBackendWithGenesisTimestamp(ChainID, 0, nil)
 	return backend, err
 }
 
-// NewL2BackendWithChainIDAndPredeploys returns a SimulatedBackend suitable for L2.
+// NewBackendWithChainIDAndPredeploys returns a SimulatedBackend suitable for L2.
 // It has the latest L2 hardforks enabled, and allows for the configuration of the network's chain ID and predeploys.
 // The returned backend should be closed after use.
-func NewL2BackendWithChainIDAndPredeploys(chainID *big.Int, predeploys map[string]*common.Address) (*backends.SimulatedBackend, error) {
-	backend, err := NewBackendWithGenesisTimestamp(chainID, 0, false, predeploys)
+func NewBackendWithChainIDAndPredeploys(chainID *big.Int, predeploys map[string]*common.Address) (*backends.SimulatedBackend, error) {
+	backend, err := NewBackendWithGenesisTimestamp(chainID, 0, predeploys)
 	return backend, err
 }
 
-func NewBackendWithGenesisTimestamp(chainID *big.Int, ts uint64, shanghai bool, predeploys map[string]*common.Address) (*backends.SimulatedBackend, error) {
-	chainConfig := params.ChainConfig{
-		ChainID:             chainID,
-		HomesteadBlock:      big.NewInt(0),
-		DAOForkBlock:        nil,
-		DAOForkSupport:      false,
-		EIP150Block:         big.NewInt(0),
-		EIP155Block:         big.NewInt(0),
-		EIP158Block:         big.NewInt(0),
-		ByzantiumBlock:      big.NewInt(0),
-		ConstantinopleBlock: big.NewInt(0),
-		PetersburgBlock:     big.NewInt(0),
-		IstanbulBlock:       big.NewInt(0),
-		MuirGlacierBlock:    big.NewInt(0),
-		BerlinBlock:         big.NewInt(0),
-		LondonBlock:         big.NewInt(0),
-		ArrowGlacierBlock:   big.NewInt(0),
-		GrayGlacierBlock:    big.NewInt(0),
-		// Activated proof of stake. We manually build/commit blocks in the simulator anyway,
-		// and the timestamp verification of PoS is not against the wallclock,
-		// preventing blocks from getting stuck temporarily in the future-blocks queue, decreasing setup time a lot.
-		MergeNetsplitBlock:            big.NewInt(0),
-		TerminalTotalDifficulty:       big.NewInt(0),
-		TerminalTotalDifficultyPassed: true,
-	}
-
-	if shanghai {
-		chainConfig.ShanghaiTime = u64ptr(0)
-	}
-
+func NewBackendWithGenesisTimestamp(chainID *big.Int, ts uint64, predeploys map[string]*common.Address) (*backends.SimulatedBackend, error) {
```

Comment:
The `chainID` seems to be unused now?

Reply:
good catch, this should be used to set chainid in the chain config. fixed.

Reply:
but this seems to have broken the tests :) looking.

Reply:
Looks like simulated backend has a fixed chainID so there's no point in letting the user specify it at all here. Removed the parameter entirely.

Reply:
Are we sure that all direct and indirect uses of `NewBackendWithGenesisTimestamp` don't require a different chain id to be set on the backend?

Reply:
Yes, there was only one indirect caller via [op-chain-ops/genesis/layer_two.go](https://github.com/ethereum-optimism/optimism/blob/5263ab95d3f463d7b79de12019ffeda3752ce33f/op-chain-ops/genesis/layer_two.go#L75)
and that has been patched to use the simulated backend chainid instead.

Reply:
My only concern is that permit2 needs to have the chain ID set at deploy time & I'd like to ensure that we're doing that correctly. There is no chainID opcode so IDK how that value is getting set, just something to pay attention to.

Reply:
There's a `CHAINID` opcode. Not sure if permit2 uses that though.

Reply:
Looks like permit2 caches the chainID as part of the constructor (deploy time): https://github.com/Uniswap/permit2/blob/main/src/EIP712.sol#L20

Reply:
Reverted the changes to the backend that removed the chain-id customization